.github/workflows/build.yaml

```diff
@@ -56,33 +73,100 @@ jobs:
       #
       # Build binary artifacts for Darwin and Linux
       #
-      # TODO: would save time if we could cache go module dependencies across build steps
       - name: Build linux binary distribution
         uses: ./.github/actions/make
         with:
           target: "release"
           os: linux
           arch: amd64
           version: ${{ steps.tag.outputs.tag }}
-      - name: Build darwin binary distribution
+      - name: Build amd64 darwin binary distribution
         uses: ./.github/actions/make
         with:
           target: "release"
           os: darwin
           arch: amd64
           version: ${{ steps.tag.outputs.tag }}
-      - name: Build M1 darwin binary distribution
+
+      - name: Build arm64 M1 darwin binary distribution
         uses: ./.github/actions/make
         with:
           target: "release"
           os: darwin
           arch: arm64
           version: ${{ steps.tag.outputs.tag }}
+
+  sign-and-notarize:
+    runs-on: macos-latest
+    needs: build
+    steps:
+      - name: Checkout current code
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.BROADBOT_TOKEN }}
+
+      - name: Pull cache
+        id: pull-cache
+        uses: actions/cache@v3
+        env:
+          cache-name: build-dir-cache
+        with:
+          path: ./output
+          key: commit-${{ github.sha }}-job-${{ github.run_id }}
+
+      # A second cache is needed to force GH to upload the new macOS tarballs,
+      # since the existing cache already exists and therefore changes made during this step
+      # are not saved for some reason. This new cache will be used during the final job
+      # where the docker image is made and the release artifacts are uploaded.
+      - name: Create release cache
+        id: release-cache
+        uses: actions/cache@v3
+        env:
+          cache-name: release-cache
+        with:
+          path: ./output/releases
+          key: commit-${{ github.sha }}-job-${{ github.run_id }}-release
+
+      - name: Create temp keychain
+        id: create-kc
+        run: |
+          mkdir -p ./output/kc
+          echo ${THELMA_MACOS_CERT} | base64 --decode > ./output/kc/certificate.p12
+          security create-keychain -p temp-kc-pwd ./output/kc/release.keychain
+          security default-keychain -s ./output/kc/release.keychain
+          security unlock-keychain -p temp-kc-pwd ./output/kc/release.keychain
+          security import ./output/kc/certificate.p12 -k ./output/kc/release.keychain -P ${THELMA_MACOS_CERT_PWD} -T /usr/bin/codesign
+          security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k temp-kc-pwd ./output/kc/release.keychain
+
+      - name: Sign and notarize macOS releases and create tarball
+        id: san-tarball
+        run: |
+          ./scripts/sign-and-notarize.sh ./output ./output/releases/thelma_${{ needs.build.outputs.version }}_darwin_amd64.tar.gz
```

Comment:
~~I have a very stupid question. Does output here contain the `arm64` or `amd64` thelma? Is it just whatever was most recently compiled? I don't understand how `sign-and-notarize` could be run in succession on the same output directory and produce results for different platforms :confused:~~

Reply:
Okay I was confused, it looks like the ./output parameter here is just a tmp directory and not input. The 2nd param is the name of the tarball to replace (so input and output)

Reply:
Correct, the first param is the output dir, so that a working dir for the script can be created inside it - the second and third are input and output tarballs.